modules/osdk-updating-projects.adoc

```diff
@@ -0,0 +1,353 @@
++
+[source,terminal]
+----
+...
+ENVTEST_K8S_VERSION = 1.24 <1>
+...
+sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.0 <2>
+----
+<1> Update version `1.22` to `1.24`.
+<2> Update version `0.7.0` to `0.9.0`.
+endif::[]
+
+.. Apply the changes to your `Makefile` and rebuild your Operator by entering the following command:
++
+[source,terminal]
+----
+$ make
+----
+
+ifdef::golang,hybrid[]
+. Make the following changes to the `go.mod` file to update Go and its dependencies:
++
+[source,golang]
+----
+go 1.18 <1>
+
+require (
+  github.com/onsi/ginkgo v1.16.5 <2>
+  github.com/onsi/gomega v1.18.1 <3>
+  k8s.io/api v0.24.0 <4>
+  k8s.io/apimachinery v0.24.0 <4>
+  k8s.io/client-go v0.24.0 <4>
+  sigs.k8s.io/controller-runtime v0.12.1 <5>
+)
+----
+<1> Update version `1.16` to `1.18`.
+<2> Update version `v1.16.4` to `v1.16.5`.
+<3> Update version `v1.15.0` to `v1.18.1`.
+<4> Update version `v0.22.1` to `v0.24.0`.
+<5> Update version `v0.10.0` to `v0.12.1`.
+endif::golang,hybrid[]
+
+ifdef::hybrid[]
+. Update the Helm Operator plug-ins:
++
+[source,golang]
+----
+github.com/operator-framework/helm-operator-plugins v0.0.11 <1>
+----
+<1> Update version `v0.0.8` to `v0.0.11`.
+endif::[]
+
+ifdef::golang,hybrid[]
+. Download and clean up the dependencies by entering the following command:
++
+[source,terminal]
+----
+$ go mod tidy
+----
+endif::[]
+
+ifdef::golang[]
+. If you use `api/webhook_suitetest.go` and `controllers/suite_test.go` suite test files, make the following changes:
++
+.Old suite test file
+[source,golang]
+----
+cfg, err := testEnv.Start()
+----
++
+.New suite test file
+[source,golang]
+----
+var err error
+// cfg is defined in this file globally.
+cfg, err = testEnv.Start()
+----
+
+. If you use the Kubernetes declarative plug-in, update your Dockerfile with the following changes:
+
+.. Add the following changes below the line that begins `COPY controllers/ controllers/`:
++
+[source,terminal]
+----
+# https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/docs/addon/walkthrough/README.md#adding-a-manifest
+# Stage channels and make readable
+COPY channels/ /channels/
+RUN chmod -R a+rx /channels/
+----
+
+.. Add the following changes below the line that begins `COPY --from=builder /workspace/manager .`:
++
+[source,terminal]
+----
+# copy channels
+COPY --from=builder /channels /channels
+----
+endif::[]
+
+ifdef::ansible[]
+. Update your `requirements.yml` file as shown in the following example:
++
+[source,yaml]
+----
+collections:
+  - name: community.kubernetes
+    version: "2.0.1" <1>
+  - name: operator_sdk.util
+    version: "0.4.0" <2>
+  - name: kubernetes.core
+    version: "2.3.1" <3>
+  - name: cloud.common
+    version: "2.1.1" <4>
+----
+<1> Update version `1.2.1` to `2.0.1`.
+<2> Update version `0.3.1` to `0.4.0`.
+<3> Update version `2.2.0` to `2.3.1`.
+<4> Add support for the Operator Ansible SDK.
```

Comment:
Are you saying that they have to add this whole cloud.common section? If so I'd clarify that and maybe move the callout number to the -name line